packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -245,9 +245,11 @@ function* doMarkChatAsRead(action: ReturnType<typeof markChatAsRead>) {
     const audiusSdk = yield* getContext('audiusSdk')
     const sdk = yield* call(audiusSdk)
     const chat = yield* select((state) => getChat(state, chatId))
-    if (!chat || !dayjs(chat?.last_read_at).isAfter(chat?.last_message_at)) {
+    if (!chat || dayjs(chat?.last_read_at).isBefore(chat?.last_message_at)) {
       yield* call([sdk.chats, sdk.chats.read], { chatId })
       yield* put(markChatAsReadSucceeded({ chatId }))
+    } else {
+      yield* put(markChatAsReadFailed({ chatId }))
```

Comment:
Nit: bit confusing bc markChatAsRead didn't actually fail, we just want to clear optimistic read count. Could define an explicit clearOptimsticReadCount fn. But honestly this is probably fine.